libs/community/langchain_community/vectorstores/vdms.py

```diff
@@ -0,0 +1,1630 @@
+            fetch_k (int): Number of candidates to fetch for knn (>= k).
+            filter (Optional[Dict[str, str]]): Filter by metadata. Defaults to None.
+
+        Returns:
+            List[Tuple[Document, float]]: List of documents most similar to
+            the query text and cosine distance in float for each.
+            Lower score represents more similarity.
+        """
+        kwargs["normalize_distance"] = True
+
+        results = self.query_collection_embeddings(
+            query_embeddings=[embedding],
+            n_results=k,
+            fetch_k=fetch_k,
+            filter=filter,
+            **kwargs,
+        )
+        return _results_to_docs_and_scores(results)
+
+    def update_document(
+        self, collection_name: str, document_id: str, document: Document
+    ) -> None:
+        """Update a document in the collection.
+
+        Args:
+            document_id (str): ID of the document to update.
+            document (Document): Document to update.
+        """
+        return self.update_documents(collection_name, [document_id], [document])
+
+    def update_documents(
+        self, collection_name: str, ids: List[str], documents: List[Document]
+    ) -> None:
+        """Update a document in the collection.
+
+        Args:
+            ids (List[str]): List of ids of the document to update.
+            documents (List[Document]): List of documents to update.
+        """
+        text = [document.page_content for document in documents]
+        metadata = [
+            validate_vdms_properties(document.metadata) for document in documents
+        ]
+        embeddings = self._embed_documents(text)
+
+        self.__update(
+            collection_name,
+            ids,
+            metadatas=metadata,
+            embeddings=embeddings,
+            documents=text,
+        )
+
+
+"""
+VDMS UTILITY
+"""
+
+
+def _results_to_docs(results: Any) -> List[Document]:
+    return [doc for doc, _ in _results_to_docs_and_scores(results)]
+
+
+def _results_to_docs_and_scores(results: Any) -> List[Tuple[Document, float]]:
+    final_res: List[Any] = []
+    responses, blobs = results[0]
+    if (
+        "FindDescriptor" in responses[0]
+        and "entities" in responses[0]["FindDescriptor"]
+    ):
+        result_entities = responses[0]["FindDescriptor"]["entities"]
+        # result_blobs = blobs
+        for ent in result_entities:
+            distance = ent["_distance"]
+            txt_contents = ent["content"]
+            for p in INVALID_DOC_METADATA_KEYS:
+                if p in ent:
+                    del ent[p]
+            props = {
+                mkey: mval
+                for mkey, mval in ent.items()
+                if mval not in INVALID_METADATA_VALUE
+            }
+
+            final_res.append(
+                (Document(page_content=txt_contents, metadata=props), distance)
+            )
+    return final_res
+
+
+def add_descriptor(
```

Comment:
Could you prefix all functions that are meant to be private with a `_` -- I assume users don't need to know what add_descriptor does